src/EditorFeatures/CSharpTest/Diagnostics/UpgradeProject/UpgradeProjectTests.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeFixes.Async;
+using Microsoft.CodeAnalysis.CSharp.UpgradeProject;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.Async
+{
+    public partial class UpgradeProjectTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddAwait)]
+        public async Task Test()
+        {
+            await TestAsync(
+@"
+class Program
+{
+    void A()
+    {
+        var x = [|(1, 2)|];
+    }
+}",
+@"
+class Program
+{
+    void A()
+    {
+        var x = (1, 2);
+    }
+}",
+new CSharpParseOptions(LanguageVersion.CSharp6));
```

Comment:
I haven't figured out how to test more than this in unittest at this point. The test helper methods don't return anything (modified project or solution) that I could inspect.

I plan to add integration tests.